frontend/src/employee-frontend/components/finance-basics/VoucherValueEditor.tsx

```diff
@@ -36,9 +42,16 @@ export default React.memo(function VoucherValueEditor({
     createVoucherValueMutation
   )
 
+  const { mutateAsync: updateVoucherValue } = useMutationResult(
+    updateVoucherValueMutation
+  )
+
   const onSubmit = useCallback(
-    () => createVoucherValue({ body: parseFormOrThrow(editorState) }),
-    [editorState, createVoucherValue]
+    () =>
+      id !== undefined
+        ? updateVoucherValue({ id: id, body: parseFormOrThrow(editorState) })
+        : createVoucherValue({ body: parseFormOrThrow(editorState) }),
```

Comment:
Throwing an uncaught exception as a part of validation is a really bad pattern. The minimum fix would be to get a rejected promise so that the user would at least see a red cross in the async button.

Reply:
I made the quick fix of returning a rejected promise, let's implement detailed validation in another PR.